core/server/api_container/server/api_container_service.go

```diff
@@ -651,3 +686,64 @@ func (apicService ApiContainerService) runStarlark(
 		}
 	}
 }
+
+func getFileDescriptionsFromArtifact(artifactPath string) ([]*kurtosis_core_rpc_api_bindings.FileArtifactContentsFileDescription, error) {
+	file, err := os.Open(artifactPath)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Failed to get file descriptions for artifact path '%v'", artifactPath)
+	}
+	defer file.Close()
+
+	gzipReader, err := gzip.NewReader(file)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Failed to get create gzip reader for artifact path '%v'", artifactPath)
+	}
+	defer gzipReader.Close()
+
+	tarReader := tar.NewReader(gzipReader)
+
+	fileDescriptions := []*kurtosis_core_rpc_api_bindings.FileArtifactContentsFileDescription{}
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
```

Comment:
Curious: could there be edge-case this happens only for just one file? but rest all the files are good and can be returned.
Maybe we could return in future {succesfulFileContents: [...], errorFileContentes:[..]}

Reply:
No, we still TARGZ one file. 1 artifact == 1 tgz, no matter the contents. This error is probably a bug in kurtosis, given that the tar is malformated